packages/parser/src/parser.ts

```diff
@@ -103,6 +103,12 @@ function parseForESLint(
     options.ecmaFeatures = {};
   }
 
+  /**
+   * Override errorOnTypeScriptSyntacticAndSemanticIssues and set it to false to prevent use from user config
+   * https://github.com/typescript-eslint/typescript-eslint/issues/8681#issuecomment-2000411834
+   */
+  options.errorOnTypeScriptSyntacticAndSemanticIssues = false;
+
```

Comment:
could you please put this down in the `Object.assign` for consistency with the other override?